08 ParamNavigation/src/components/member/memberPage.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { hashHistory } from 'react-router'
+import * as toastr from 'toastr';
+import MemberEntity from './../../api/memberEntity'
+import MemberForm from './memberForm';
+import MemberAPI from '../../api/memberAPI';
+//import * as ObjectAssign from 'object-assign';
+import objectAssign = require('object-assign');
+
+interface Props extends React.Props<memberPage> {
+  params : any
+}
+
+interface State {
+  member : MemberEntity
+  ,errors: any
+  ,dirty  : boolean
+}
+
+// Nice tsx guide: https://github.com/Microsoft/TypeScript/wiki/JSX
+export default class memberPage extends React.Component<Props, State> {
+
+  constructor(props : Props){
+        super(props);
+        // set initial state
+        this.state = {
+                      member: new MemberEntity()
+                      ,errors : {}
+                      ,dirty : false
+                     };
+  }
+
+  componentWillMount() {
```

Comment:
componentWillMount() or componentDidMount()?

> **Mounting: componentWillMount**
> void componentWillMount()
> Invoked once, both on the client and server, immediately before the initial rendering occurs. If you call setState within this method, render() will see the updated state and will be executed only once despite the state change.


> **Mounting: componentDidMount #**
> void componentDidMount()
> Invoked once, only on the client (not on the server), immediately after the initial rendering occurs. At this point in the lifecycle, you can access any refs to your children (e.g., to access the underlying DOM representation). The componentDidMount() method of child components is invoked before that of parent components.
> 
> If you want to integrate with other JavaScript frameworks, set timers using setTimeout or setInterval, or send AJAX requests, perform those operations in this method.

[https://facebook.github.io/react/docs/component-specs.html](https://facebook.github.io/react/docs/component-specs.html)

I think the key is in 

> or send AJAX requests